### PR TITLE
[INTERNAL] Move Transfer Listener Logic

### PR DESCRIPTION
**Pull request by stefaus**
In preparation for the new feature 'Instant Session Start', this patch
move parts of the Transfer Listener logic in Archive type to the Abstract
class and prevent later code duplication.

### CODE REVIEW

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/AbstractIncomingProjectNegotiation.java [hunk 1]
```diff
@@ -184,8 +188,16 @@ public Status run(Map<String, IProject> projectMapping,
      *
      * @throws IOException, SarosCancellationException
      */
-    protected abstract void setup(IProgressMonitor monitor)
-        throws IOException, SarosCancellationException;
+    protected void setup(IProgressMonitor monitor) throws IOException,
```
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/AbstractIncomingProjectNegotiation.java]: Code Smell Code Smell: Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. ([squid:RedundantThrowsDeclarationCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3ARedundantThrowsDeclarationCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=127&issues=AWSQldZGMDLKieESChXO&open=AWSQldZGMDLKieESChXO)

[ISSUE_KEY]: AWSQldZGMDLKieESChXO

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/AbstractIncomingProjectNegotiation.java [hunk 2]
```diff
@@ -220,6 +232,16 @@ protected void cleanup(IProgressMonitor monitor,
         Map<String, IProject> projectMapping) {
         fileReplacementInProgressObservable.replacementDone();
 
+        /*
+         * TODO Move disable queuing responsibility to SarosSession (see todo
```
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/AbstractIncomingProjectNegotiation.java]: Code Smell Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=127&issues=AWSQldZGMDLKieESChXP&open=AWSQldZGMDLKieESChXP)

[ISSUE_KEY]: AWSQldZGMDLKieESChXP

Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/AbstractIncomingProjectNegotiation.java]: Code Smell Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=127&issues=AWSQldZGMDLKieESChXP&open=AWSQldZGMDLKieESChXP)

[ISSUE_KEY]: AWSQldZGMDLKieESChXP

Comment (stefaus) [reply de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/AbstractIncomingProjectNegotiation.java]: Handle in #137

Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/AbstractIncomingProjectNegotiation.java]: Code Smell Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=127&issues=AWSQldZGMDLKieESChXP&open=AWSQldZGMDLKieESChXP)

[ISSUE_KEY]: AWSQldZGMDLKieESChXP

**Comment by stefaus** [comment]
@srossbach Are you still requesting a change? To explain my point further, I link to Martin Fowler's post about [Yagni](https://martinfowler.com/bliki/Yagni.html).

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/AbstractIncomingProjectNegotiation.java (hunk 2 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/AbstractIncomingProjectNegotiation.java]: Code Smell Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=127&issues=AWSQldZGMDLKieESChXP&open=AWSQldZGMDLKieESChXP)

[ISSUE_KEY]: AWSQldZGMDLKieESChXP

**Pull request merged**